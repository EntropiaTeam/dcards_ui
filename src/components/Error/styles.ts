import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
  errorContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexFlow: 'column',
    height: '100%'
  },
  errorText: {
    fontSize: 16,
    textAlign: 'center'
  },
  buttonContainer: {
    marginTop: 16
  }
});

export {
  useStyles
};
