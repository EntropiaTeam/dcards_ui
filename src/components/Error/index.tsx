import React, { FC } from 'react';
import Box from '@material-ui/core/Box';
import MainButton from '../UI/MainButton';
import { useStyles } from './styles';

interface ErrorProps {
  errorText: string;
  retryCallback?: () => void;
}

const Error: FC<ErrorProps> = ({ errorText, retryCallback }) => {
  const classes = useStyles();
  return (
    <Box className={classes.errorContainer}>
      <Box component="span" className={classes.errorText}>{errorText}</Box>
      {retryCallback && (
      <Box className={classes.buttonContainer}>
        <MainButton onClick={retryCallback}>Retry</MainButton>
      </Box>
      )}
    </Box>
  );
};

export default Error;
