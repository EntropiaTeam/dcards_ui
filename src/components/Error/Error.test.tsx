import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Error from '.';
import { MainButton } from '..';

describe('<Error />', () => {
  const mockCallback = jest.fn();

  const props = {
    errorText: 'The error is occured!',
    retryCallback: mockCallback
  };

  const ErrorComponent = renderer.create(<Error {...props} />).toJSON();

  const wrapper = mount(<Error {...props} />);

  it('should render Error component correctly', () => {
    expect(ErrorComponent).toMatchSnapshot();
  });

  it('should render Error component with proper errorText prop', () => {
    expect((wrapper).prop('errorText')).toEqual(props.errorText);
  });

  it('should render one MainButton component with proper text', () => {
    expect(wrapper.find(MainButton)).toHaveLength(1);
    expect(wrapper.find(MainButton).text()).toEqual('Retry');
  });

  it('should simulate click to invoke retry callback', () => {
    wrapper.find(MainButton).simulate('click');
    expect(mockCallback.mock.calls.length).toEqual(1);
    expect(props.retryCallback).toHaveBeenCalledTimes(1);
  });
});
