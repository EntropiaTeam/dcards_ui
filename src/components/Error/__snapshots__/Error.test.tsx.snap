// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render Error component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-errorContainer-1"
>
  <span
    className="MuiBox-root MuiBox-root-5 makeStyles-errorText-2"
  >
    The error is occured!
  </span>
  <div
    className="MuiBox-root MuiBox-root-6 makeStyles-buttonContainer-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary makeStyles-primary-8 MuiButton-disableElevation"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Retry
      </span>
    </button>
  </div>
</div>
`;
